tests/bofire/kernels/test_categorical.py

```diff
@@ -0,0 +1,18 @@
+import torch
+from botorch.models.kernels.categorical import CategoricalKernel
+from botorch.models.transforms.input import OneHotToNumeric
+
+from bofire.kernels.categorical import HammingKernelWithOneHots
+
+
+def test_hamming_with_one_hot():
```

Comment:
I am not 100% sure that this test really is complete enough, as you are only testing here for one categorical feature, furthermore you are testing with `ard_num_dims=1` which is the default, and uses one lenthscale for everything. In this case, by coincidence one gets the same results. 

But I would already expect that the HammingDistanceKernel only works with `ard_num_dims=1` or three. As it treats in the current setup every one-hot expanded column as an independet feature.